m0t1fs/linux_kernel/file.c

```diff
@@ -3348,6 +3348,42 @@ static void dgmode_rwvec_dealloc_fini(struct dgmode_rwvec *dg)
 	m0_varr_fini(&dg->dr_pageattrs);
 }
 
+static void databufs_set_dgw_mode(struct pargrp_iomap *iomap,
+				  struct m0_ext *ext)
+{
+	uint32_t         row_start;
+	uint32_t         row_end;
+	uint32_t         row;
+	uint32_t         col;
+	struct data_buf *dbuf;
+
+	page_pos_get(iomap, ext->e_start, &row_start, &col);
```

Comment:
There is a kernel based implementation of page_pos_get() here in m0t1fs/linux_kernel/file.c

There is user space implementation of page_pos_get() defined in motr/utils.c
They have different parameter.
But I think the user-space implementation can be changed to have the same parameters.

Reply:
Yes, there is a lot of duplicated code in our kernel and user-space client I/O implementation. It's a bit task to merge them, which is out of scope of this modest PR. :)